### PR TITLE
Evany ioerror fix

### PR DESCRIPTION
**Pull request by yangev**
Applies this patch: https://github.com/prometheus/client_python/issues/425, but more broadly, since we also archive existing metrics

### CODE REVIEW

File: prometheus_client/multiprocess.py [hunk 1]
```diff
@@ -51,17 +51,20 @@ def merge(files, accumulate=True):
             pid = parts[2] if multiprocess_mode and len(parts) > 2 else None
             try:
                 d = MmapedDict(f, read_mode=True)
-            except FileNotFoundError:
+            except EnvironmentError:
```
Comment (yangev) [inline prometheus_client/multiprocess.py:54]: EnvironmentError being both python2 and python3 compatible, as opposed to FileNotFoundError and IOError

**Comment by yangev** [comment]
My main concern is that I'm not exactly clear how counter resets work

File: prometheus_client/multiprocess.py [hunk 2]
```diff
@@ -49,7 +49,22 @@ def merge(files, accumulate=True):
...
+                # boot.gunicornconf.child_exit).
+                # Additionally, we have a single thread which will collect
+                # metrics files from dead workers, and merge them into a set of
+                # archive files at regular interviews (see
+                # multiprocess_exporter).
+                # Since collecting the files to
+                # merge and reading those files are non-atomic, it's very
+                # possible, and expected, that these files will not exist at
+                # this point
+                continue
```
Comment (cvanderschuere) [inline prometheus_client/multiprocess.py:67]: What other metrics come through here that we would want to raise on?

https://github.com/prometheus/client_python/blob/9df82be86a49130c3093b961f88955825c471a2a/prometheus_client/multiprocess.py#L62-L68

**Review by cvanderschuere — APPROVED** [review]
Would be nice to have some more commentary about how/why our change deviates from upstream, but otherwise LGTM

**Pull request merged**